proposals/lip-0048.md

```diff
@@ -25,15 +25,57 @@ This LIP defines the fee system in a modular way, as currently [used in the Lisk
 
-Chains can choose that this minimum fee gets either burned (to avoid that validators can send transactions in the blocks they generate without cost) or transferred to the specified pool address `ADDRESS_FEE_POOL`. The latter choice could be beneficial for sidechains, as fees collected in this way can be used as an incentive for validators and other protocol participants. In the Lisk mainchain, the minimum fee is always burned.
+#### Minimum Fee per Transaction
+
+As introduced in [LIP 0013][lip-0013], all transactions must have a transaction fee that is at least equal to a *minimum fee*. This fee corresponds to the cost of using blockspace and is therefore dependent on the size of the transaction, with the proportionality constant of `MIN_FEE_PER_BYTE`: 
+```python
+trsSize = length(encodeTransaction(trs))
+minFee = MIN_FEE_PER_BYTE * trsSize
+```
+The minimum fee is always subtracted from the part of the fee that goes to the block generator, as we explain below. Therefore, the minimum fee discourages validators from sending zero-cost transactions in the blocks they generate.
+
+Note that chains can freely configure `MIN_FEE_PER_BYTE`, which can even be set to zero. The `MIN_FEE_PER_BYTE` value cannot be changed, unless a hard fork occurs. For this reason, we allow sidechains to define a period of `MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE` blocks after the genesis block, for which the minimum fee does not apply. 
+
+This could, for example, be beneficial for sidechains that want to use LSK as the fee token. Initially, there is no LSK on the sidechain, so the sidechain cannot process transactions unless `MIN_FEE_PER_BYTE` is set to zero, which is generally not desirable since it essentially disables the feature of minimum fee for the chain. By granting an initial exemption period of `MAX_BLOCK_HEIGHT_ZERO_FEE_PER_BYTE` blocks, users have a window of time to transfer LSK tokens to the sidechain, after which the initially set value of `MIN_FEE_PER_BYTE` would start being applied. 
+
+### Additional Fees
+
+In principle, any transaction that results in a state increase or significant computational effort could require a fee. Since the current implementation does not provide an automatic way to charge fees in these cases, we introduce the [`payFee`](#payfee) method. 
+
+This method can be invoked by other modules to charge an additional fee when a state increase is triggered during the execution of a transaction, e.g., when a new validator is registered in the PoS module, a new sidechain is registered, or a [user account is initialized][lip-0051#initializeuseraccountinternal] in the Token module: 
+```python
+Fee.payFee(USER_ACCOUNT_INITIALIZATION_FEE)
+```
+In addition, developers may decide to call the `payFee` method for transactions that require extensive computation, such as when verifying an inclusion proof.
```

Comment:
I would add something like: "..., or if any other resource is extensively used during transaction execution" to be slightly more general. Maybe a transaction accesses a decentralized storage or interacts with an oracle, etc.

Reply:
Right. Changed as suggested.